### PR TITLE
Improvements to the Register Allocator #80 

### PR DESCRIPTION
**Pull request by tgiphil**
Avoids register spills to stack in certain use-cases (#80), and better register selection.

Results:

* 27% less spill to the stack
* 3% less register moves 

Also include fixes register size issue when inlining methods.

### CODE REVIEW

**Comment by tgiphil** [comment]
Sorry; lots of code clean up and package updates as well.

**Review by charsleysa — CHANGES_REQUESTED** [review]

File: Source/Mosa.Compiler.Framework/Managers/BitValueManager.cs [hunk 1]
```diff
@@ -50,19 +49,65 @@ public BitValue GetBitValue(Operand operand)
...
+			return (BitValue.Any64, BitValueStage.Stable);
+		}
+		else
+		{
+			return (BitValue.Any32, BitValueStage.Stable);
+		}
 	}
 
-	public void UpdateBitValue(Operand operand, BitValue value)
+	//public BitValue GetBitValueDefaultAny(Operand operand)
```
[the remark this answers is not in the document]
Comment (tgiphil) [reply Source/Mosa.Compiler.Framework/Managers/BitValueManager.cs:98]: Yes; it's going away soon.

File: Source/Mosa.Compiler.Framework/TransformContext.cs [hunk 2]
```diff
@@ -469,10 +469,10 @@ public BitValue GetBitValue(Operand operand)
 		return BitValueManager.GetBitValue(operand);
 	}
 
-	public BitValue GetBitValueWithDefault(Operand operand)
-	{
-		return BitValueManager.GetBitValueDefaultAny(operand);
-	}
+	//public BitValue GetBitValueWithDefault(Operand operand)
```
[the remark this answers is not in the document]
Comment (tgiphil) [reply Source/Mosa.Compiler.Framework/TransformContext.cs:472]: Yes; it's going away soon.

**Review by charsleysa — APPROVED** [review]
LGTM

**Pull request merged**